tests/autologging/fixtures.py

```diff
@@ -31,3 +49,30 @@ def test_mode_on():
             os.environ[_AUTOLOGGING_TEST_MODE_ENV_VAR] = prev_env_var_value
         else:
             del os.environ[_AUTOLOGGING_TEST_MODE_ENV_VAR]
+
+
+@pytest.fixture(autouse=True)
+def reset_stderr():
+    prev_stderr = sys.stderr
+    yield
+    sys.stderr = prev_stderr
+
+
+@pytest.fixture(autouse=True)
+def reset_logging_enablement():
+    yield
+    logging_utils.enable_logging()
+
+
+class TestStream:
```

Comment:
This works but have you considered standard options like io.StringIO? Inheriting from io.IOBase and override the class method there could also indicate a more explicit intent.